modules/state_machine/states/dash.gd

```diff
@@ -40,7 +40,10 @@ func handle_input(actor, event):
 func process(actor, delta):
 	actor.velocity.y = 0
 	if actor.is_on_wall():
-		actor.stop()
+		if actor.is_on_floor():
+			actor.stop()
+		else:
+			actor.wall_slide()
```

Comment:
This only makes the correct transition from the dash state depending on either the character is in air or on floor